mro/__init__.py

```diff
@@ -155,59 +131,94 @@ def none_transform(column_default, data_type):
             if column_default:
                 column_default, get_value_on_insert = data_type[2](column_default, column[7])
 
-            code_start = 'data_types.{}("{}", {}, '.format(data_type[0], column_name, column_index)
-            code_end = 'not_null={0}, is_updatable={1}, get_value_on_insert={2}, is_primary_key={3})'.format(not is_nullable, is_updateable, get_value_on_insert, is_primary_key)
-
-            code = data_type[1](column, code_start, code_end)
+            col_data['data_type'] = data_type[0]
+            col_data['column_name'] = column_name
+            col_data['column_index'] = column_index
+            col_data['column_default'] = column_default
+            col_data['not_null'] = not is_nullable
+            col_data['is_updateable'] = is_updateable
+            col_data['get_value_on_insert'] = get_value_on_insert
+            col_data['is_primary_key'] = is_primary_key
+            col_data['length'] = column[8]
             if column_name in foreign_keys:
                 foreign_key = foreign_keys[column_name]
-                # passing class name as string for eval to get around creation order issues
-                code = 'mro.foreign_keys.foreign_key_data_type("{}", {}, "mro.{}", "{}")'.format(column_name, code, foreign_key[0], foreign_key[1])
-            col_data.append((code, column_name, column_default))            
-
-        for foreign_key_target in foreign_key_targets:
-            foreign_key_name = foreign_key_target[0] + 's'
-            # if they happen to have a column the same name as the refernece list don't add it
-            if foreign_key_name not in [c[1] for c in col_data]:
-                code = 'mro.foreign_keys.foreign_key_reference("{}", "mro.{}", "{}")'.format(foreign_key_target[2], foreign_key_target[0], foreign_key_target[1])
-                col_data.append((code, foreign_key_target[0] + 's', None))
+                col_data['foreign_key'] = foreign_key
 
-        tables[table_name] = col_data
+            columns.append(col_data)
+        tables[table_name] = {}
+        tables[table_name]['columns'] = columns
+        tables[table_name]['foreign_key_targets'] = foreign_key_targets
 
-    return tables   
+    return tables
 
 
 def _create_classes(tables):
-    for table, columns in tables.items():
-        code = 'class ' + table + '(mro.table.table):\n'
-        for column_data in columns:
-            code += '    ' + column_data[1] + '=' + column_data[0] + '\n'
-        code += """    def __init__(self, **kwargs):\n"""
```

Comment:
this has changed to use python code instead of a large string format + exec, this allows you to debug into it.